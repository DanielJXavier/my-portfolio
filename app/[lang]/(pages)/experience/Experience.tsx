"use client";

import { createContext, useRef, useState } from "react";

import { useParams } from "next/navigation";

import { ExperienceItemsType, getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import Title from "@/_components/Title";
import Work from "@/_icons/Work";

import Menu from "../_components/Menu";

import ExperienceItem from "./_components/ExperienceItem";
import { experience } from "./_config";

export const ExperienceModeContext = createContext("");

export default function Experience() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: { title, menu, items, imageAltText },
  } = getDictionary(lang);

  const [experienceMode, setExperienceMode] = useState("simple");

  const menuItems = useRef([
    { text: menu.simple, value: "simple" },
    { text: menu.detailed, value: "detailed" },
  ]);

  const experienceItems = useRef<ExperienceItemsType>(items);

  return (
    <ExperienceModeContext.Provider value={experienceMode}>
      <Title icon={<Work />}>{title}</Title>

      <Menu
        items={menuItems.current}
        activeItem={experienceMode}
        handleClick={setExperienceMode}
      />

      {experience
        .filter(({ hidden }) => !hidden)
        .map(
          ({
            key,
            companyId,
            companyName,
            startDate,
            endDate,
            mainStack,
            hasBlackLogo,
          }) => (
            <ExperienceItem
              key={key}
              role={experienceItems.current[key].role}
              companyId={companyId}
              companyName={companyName}
              startDate={startDate}
              endDate={endDate}
              accomplishments={experienceItems.current[key].accomplishments}
              mainStack={mainStack}
              imageAltText={imageAltText}
              hasBlackLogo={hasBlackLogo}
            />
          )
        )}
    </ExperienceModeContext.Provider>
  );
}
