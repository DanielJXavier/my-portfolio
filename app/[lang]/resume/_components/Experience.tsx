import { useParams } from "next/navigation";

import { ExperienceItemsType, getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { experience } from "@/[lang]/(pages)/experience/_config";
import { hardSkills } from "@/[lang]/(pages)/skills/_config";

import VisibilityControl from "./VisibiltyControl";
import { useRef } from "react";

export default function Experience() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: {
      items,
      previousItemsText,
      biggestChallengeTitle,
      mainStackTitle,
    },
    resume: { sectionTitles },
  } = getDictionary(lang);

  const experienceItems = useRef<ExperienceItemsType>(items);

  return (
    <section>
      <h2 className="mb-2 text-[16pt] font-bold">{sectionTitles.experience}</h2>
      <ul className="flex flex-col gap-y-3">
        {experience.map(
          ({
            key,
            companyName,
            year,
            mainStack,
            resumeAccomplishments,
            resume,
          }) => (
            <VisibilityControl
              key={key}
              className="items-baseline break-inside-avoid"
              defaultVisible={resume}
            >
              <div>
                <p className="text-[12pt]">
                  <span className="font-bold">
                    {experienceItems.current[key].role} @ {companyName}
                  </span>
                  <span> ({year})</span>
                </p>
                <ul className="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['â€¢'] [&_li]:before:absolute [&_li]:before:-left-[10px]">
                  {experienceItems.current[key].accomplishments.map(
                    (accomplishment, i) => (
                      <VisibilityControl
                        key={`accomplishment-${i}`}
                        className="items-baseline"
                        defaultVisible={resumeAccomplishments?.some(
                          (resumeAccomplishment) =>
                            resumeAccomplishment === i + 1
                        )}
                      >
                        {accomplishment}
                      </VisibilityControl>
                    )
                  )}
                  <VisibilityControl
                    className="mt-1 items-baseline"
                    defaultVisible={resume}
                  >
                    {biggestChallengeTitle}:{" "}
                    {experienceItems.current[key].biggestChallenge}
                  </VisibilityControl>
                  <VisibilityControl
                    className="mt-1 items-baseline"
                    defaultVisible={resume}
                  >
                    {mainStackTitle}:{" "}
                    {hardSkills
                      .filter(({ key }) =>
                        mainStack.some((stack) => stack === key)
                      )
                      .map(({ name }) => name)
                      .join(" | ")}
                  </VisibilityControl>
                </ul>
              </div>
            </VisibilityControl>
          )
        )}
        <li>
          {previousItemsText.prefix} {previousItemsText.sufix}
        </li>
      </ul>
    </section>
  );
}
