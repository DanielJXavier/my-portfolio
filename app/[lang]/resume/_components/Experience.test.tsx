import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Experience from "./Experience";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Experience component", () => {
  it("renders the component", () => {
    const { container } = render(<Experience />);

    expect(container).toMatchSnapshot();
  });
});
