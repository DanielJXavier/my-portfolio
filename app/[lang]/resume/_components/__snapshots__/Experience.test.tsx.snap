// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience component renders the component 1`] = `
<div>
  <section>
    <h2
      class="text-[16pt] font-bold"
    >
      Experience
    </h2>
    <ul
      class="flex flex-col gap-y-3"
    >
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Software Engineering Manager
             @ 
            OLX
          </p>
          <p
            class="text-[12pt]"
          >
            2024
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led a team of 9 developers, serving as a mentor and guiding light;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Delivered the successful execution of the platform's registration and ad management system roadmap including the new Ad Insertion in steps journey;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Built a high level of engagement within the team;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Held personalized "1:1s" to guide team members through their performance cycles;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Adopted agile methodologies and led team ceremonies to ensure a streamlined workflow balancing product and technical tasks;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Worked closely with the Product Manager to facilitate informed and effective product decisions;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To deal with pressure from leadership to do major deliveries in the short term.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Software Engineering Manager
             @ 
            OLX
          </p>
          <p
            class="text-[12pt]"
          >
            2023
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led a team of 8 developers, serving as a mentor and guiding light;¹
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Delivered the successful execution of the search and recommendation systems roadmap including the new LTR rank algorithm;¹
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Structured the team and proposed a new approach to support the strategic plan for the upcoming year;¹
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Clarified the company's understanding of search and recommendation subjects;¹
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Supervised all developers (9 devs) in the chapter indirectly, fostered a sense of unity and direction;²
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Fostered engagement within the chapter and ensured valuable deliveries despite working only once a week;²
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To balance the substantial responsibilities of both the team¹ and the chapter² simultaneously.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Software Engineering Manager
             @ 
            OLX
          </p>
          <p
            class="text-[12pt]"
          >
            2022 ~ 2023
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led a team of 9 developers, serving as a point of reference;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Partnered with the Design Manager to make optimal product decisions and establish a roadmap;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Delivered the Design System roadmap including an entirely new platform homepage created in less than 48 hours;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Gathered and analyzed data to validate the project's value to the company;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Ensured the continuous evolution of the Design System and its adoption by other teams;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed documentation and processes for contributing to the Design System project and managing its version;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To assume multiple roles simultaneously, including Engineering Manager, Product Manager, Data Analyst, among others.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Software Engineering Manager
             @ 
            OLX
          </p>
          <p
            class="text-[12pt]"
          >
            2021 ~ 2022
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led a team of 7 developers, serving as a key reference;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Delivered both direct and indirect Advertising roadmaps for the platform that brought R$80 million in revenue;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Recruited new professionals to increase productive capacity in 300%;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Partnered actively with the Product Manager to prioritize and develop optimal initiatives;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Proposed a new structure for the Advertising area to enhance results;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To manage high expectations related to Advertising revenue results.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            CTO & Partner
             @ 
            Meaning
          </p>
          <p
            class="text-[12pt]"
          >
            2023
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Crafted the startup's technical vision and product strategy;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Architected the solution from the ground up;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Refined the product's technical specifications;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed a Proof of Concept project;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To convert a highly "human-centric" project, rooted in personal consultancy, into a systematic approach.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Development Coordinator
             @ 
            Softplan
          </p>
          <p
            class="text-[12pt]"
          >
            2020 ~ 2021
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led directly a multidisciplinary software development team, inclusive of Software Architects, Developers, Quality Assurance Analysts, Product Owners, Business Analysts, and Agile Masters;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Mentored and developed both direct and indirect employees, utilizing techniques such as 1:1 and feedforward;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Planned and executed deliveries for the most 2 critical modules of the "SAJ" system;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Applied daily situational leadership to maximize the potential of those under my guidance;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Structured and continuously enhanced the team's workflow using agile software development methodologies;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Created formulas and metrics to accurately gauge the team's capacity, thereby bringing predictability to deliveries;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Mapped effective talent retention strategies;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To manage a large team (28 professionals) and oversee critical deliveries.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Tech Lead
             @ 
            Softplan
          </p>
          <p
            class="text-[12pt]"
          >
            2019 ~ 2020
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Developed the new "SAJ" system, constructed using React JS, .Net, and GraphQL;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Oversaw the deliveries and deployment of 17 modules of the system;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Provided training in HTML, CSS, JavaScript, React JS, GraphQL, and unit tests;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Led a team of 10 Front-End React developers;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Recruited new employees for the company;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Selected new contractor professionals;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed professionals through 1:1 sessions, Feedforwards, and Mentoring;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To acquire leadership skills and master people management.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Senior Front-End Developer
             @ 
            Softplan
          </p>
          <p
            class="text-[12pt]"
          >
            2018 ~ 2019
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Developed new features for the "e-SAJ" portal utilizing React JS and GraphQL;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Created new features for the "SAJ Tribunais" system using React JS and GraphQL;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 "
              data-testid="visibility-control"
              data-visible=""
            >
              <input
                checked=""
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To manage all the deliveries of the project as the sole Front-End developer.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
        data-testid="visibility-control"
      >
        <input
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Senior Front-End Developer
             @ 
            Hotmart
          </p>
          <p
            class="text-[12pt]"
          >
            2018
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed systems to address the company's internal needs and automate tasks, leveraging technologies such as Vue.js, and Node.js;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Created a Proof of Concept (POC) that introduced a simplified method for product registration on the platform;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To balance the need to comprehend customer requirements clearly while simultaneously coding.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
        data-testid="visibility-control"
      >
        <input
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Mid Level Front-End Developer
             @ 
            Hotmart
          </p>
          <p
            class="text-[12pt]"
          >
            2017 ~ 2018
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Designed a system to generate blog post images using components and pre-defined templates, thereby enhancing the productivity of the design team by 1000%;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed a high-performance caching system for the "Hotmart blog" (WordPress), which resulted in a 300% increase in performance;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Implemented the Arabic version of the "Hotmart blog";
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To conceptualize and develop complex solutions from the ground up.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
        data-testid="visibility-control"
      >
        <input
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Entry Level Front-End Developer
             @ 
            Hotmart
          </p>
          <p
            class="text-[12pt]"
          >
            2016 ~ 2017
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Developed a new WordPress theme for the "Hotmart blog" utilizing only HTML5, CSS Grid Layout, and Vanilla JS;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Implemented the new layout for the "Hotmart blog";
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Created custom features within the WordPress admin interface;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To acquire new development skills and apply them to complex projects.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
        data-testid="visibility-control"
      >
        <input
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Support Analyst
             @ 
            Sydle
          </p>
          <p
            class="text-[12pt]"
          >
            2014 ~ 2016
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Spearheaded the development of web projects utilizing WordPress and Magento;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Facilitated the modeling and automation of the company's internal processes through BPMN;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Provided comprehensive technical support to users of the company's systems;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Oversaw the configuration and monitoring of both Linux and Windows servers.
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To rapidly acquire development skills and contribute to legacy projects.
            </li>
          </ul>
        </div>
      </li>
      <li
        class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
        data-testid="visibility-control"
      >
        <input
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Support Intern
             @ 
            Sydle
          </p>
          <p
            class="text-[12pt]"
          >
            2012 ~ 2014
          </p>
          <ul
            class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['•'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
          >
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Provided technical support to users of the company's systems;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Configured and monitored infrastructure on Linux and Windows servers;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Compiled team reports;
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Offered infrastructure support to the company's employees.
            </li>
            <li
              class="items-baseline flex gap-x-1.5 opacity-50 line-through print:hidden"
              data-testid="visibility-control"
            >
              <input
                class="print:hidden"
                type="checkbox"
              />
              Biggest challenge
              :
               
              To simultaneously learn a multitude of new concepts and balance them with my university commitments.
            </li>
          </ul>
        </div>
      </li>
      <li>
        ... and
         
        other previous roles
      </li>
    </ul>
  </section>
</div>
`;
